### PR TITLE
Added an explanation on how to pass data to the script via data-mage-init and to access them in the script.

### PR DESCRIPTION
**Pull request by sathiyaaa**
## Purpose of this pull request

This pull request (PR)  covers #8256 

## Affected DevDocs pages

- /guides/v2.4/javascript-dev-guide/javascript/requirejs.html

whatsnew
Added an [explanation and example](http://devdocs.magento.com/guides/v2.4/javascript-dev-guide/javascript/requirejs.html) of passing parameters to JavaScript using the `data-mage-init` HTML attribute.

### CODE REVIEW

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 1]
```diff
@@ -31,6 +31,65 @@ In Magento 2, you have two options for specifying declarative notation:
...
+        <div class="item">Item 1</div>
+        ...
+        <div class="item">Item n</div>
+    </div>
+
+    <script type="text/x-magento-init">
+        {
+            "#<carousel_name>": {
+            "carousel": {"option": value}
+        }
```
Comment (BarnyShergold) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:88]: This block is not indented properly - braces should never be one on top of another

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Nice addition - just need some formatting issues sorted

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 2]
```diff
@@ -31,6 +31,61 @@ In Magento 2, you have two options for specifying declarative notation:
...
+    <div id="<carousel_name>" class="carousel">
+        <div class="item">Item 1</div>
+        ...
+        <div class="item">Item n</div>
+    </div>
+
+    <script type="text/x-magento-init">
+        {
+            "#<carousel_name>":
+            {
```
Comment (BarnyShergold) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:81]: Brace should be on previous line

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 3]
```diff
@@ -31,6 +31,61 @@ In Magento 2, you have two options for specifying declarative notation:
...
+            "carousel": {"option": value}
```
Comment (BarnyShergold) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:82]:
```suggestion
                "carousel": {"option": value}
```

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 4]
```diff
@@ -31,6 +31,61 @@ In Magento 2, you have two options for specifying declarative notation:
...
+        {
+            $(element).<carousel_name>(config);
+        };
+   });
+   ```
+-  Add RequireJS config to `app/design/frontend/<package_name>/<theme_name>/requirejs-config.js`
+
+    ```javascript
+    var config = {
+    map: {
```
Comment (BarnyShergold) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:51]: Map section needs to be indented a further level

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 5]
```diff
@@ -31,6 +31,61 @@ In Magento 2, you have two options for specifying declarative notation:
...
+        '*': {
+            'carousel': 'js/carousel',
```
Comment (BarnyShergold) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:53]: Contents of this brace enclosure need to be indented another level

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 6]
```diff
@@ -31,6 +31,61 @@ In Magento 2, you have two options for specifying declarative notation:
 
    > This is used to target either a CSS selector or `*`. If the CSS selector matches multiple HTML elements, the script will run for each matched HTML element. For `*`, no HTML element is selected and the script will run once with the HTML DOM as its target. This method can be implemented from anywhere in the codebase to target any HTML element. This is preferred when direct access to the HTML element is restricted, or when there is no target HTML element.
 
+Consider the example of adding a custom carousel JS
```
Comment (BarnyShergold) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:34]:
```suggestion
Consider the example of adding a custom carousel JS:
```

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 7]
```diff
@@ -31,6 +31,61 @@ In Magento 2, you have two options for specifying declarative notation:
...
+    var config = {
+    map: {
+        '*': {
+            'carousel': 'js/carousel',
+            '<carousel_name>': 'js/<carousel_name>/<carousel_name>.carousel'
+             }
+        }
+    };
+    ```
+ You now have two options for specifying declarative notation:
```
Comment (BarnyShergold) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:59]: Should be a blank line between list and previous sentence

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Looking a lot cleaner - a few format errors left

Also you have some linting errors - please correct - they are trailing space issues

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 8]
```diff
@@ -31,6 +31,61 @@ In Magento 2, you have two options for specifying declarative notation:
 
    > This is used to target either a CSS selector or `*`. If the CSS selector matches multiple HTML elements, the script will run for each matched HTML element. For `*`, no HTML element is selected and the script will run once with the HTML DOM as its target. This method can be implemented from anywhere in the codebase to target any HTML element. This is preferred when direct access to the HTML element is restricted, or when there is no target HTML element.
 
+Consider the example of adding a custom carousel JS:
+
+-  Copy `<carousel_name>.carousel.js` to `app/design/frontend/<package_name>/<theme_name>/web/js/<carousel_name>/`
+-  Add your RequireJS module at `app/design/frontend/<package_name>/<theme_name>/web/js/carousel.js`
+   ```javascript
```
Comment (BarnyShergold) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:38]: Be good to have a blank line here (as below)

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
One last small suggestion! All good after that

**Review by BarnyShergold — APPROVED** [review]
All good! Nice work.

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 9]
```diff
@@ -31,6 +31,62 @@ In Magento 2, you have two options for specifying declarative notation:
 
    > This is used to target either a CSS selector or `*`. If the CSS selector matches multiple HTML elements, the script will run for each matched HTML element. For `*`, no HTML element is selected and the script will run once with the HTML DOM as its target. This method can be implemented from anywhere in the codebase to target any HTML element. This is preferred when direct access to the HTML element is restricted, or when there is no target HTML element.
 
+Consider the example of adding a custom carousel JS:
+
+-  Copy `<carousel_name>.carousel.js` to `app/design/frontend/<package_name>/<theme_name>/web/js/<carousel_name>/`
```
Comment (jeff-matthews) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:36]:
```suggestion
1. Copy the `<carousel_name>.carousel.js` file to the `app/design/frontend/<package_name>/<theme_name>/web/js/<carousel_name>/` directory.
```

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 10]
```diff
@@ -31,6 +31,62 @@ In Magento 2, you have two options for specifying declarative notation:
...
+-  Add your RequireJS module at `app/design/frontend/<package_name>/<theme_name>/web/js/carousel.js`
```
Comment (jeff-matthews) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:37]:
```suggestion
1. Add your RequireJS module at `app/design/frontend/<package_name>/<theme_name>/web/js/carousel.js`.
```

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 11]
```diff
@@ -31,6 +31,62 @@ In Magento 2, you have two options for specifying declarative notation:
...
+   ```javascript
+    define(['jquery','<carousel_name>'], function($)
+    {
+        return function(config, element)
+        {
+            $(element).<carousel_name>(config);
+        };
+    });
+   ```
+-  Add RequireJS config to `app/design/frontend/<package_name>/<theme_name>/requirejs-config.js`
```
Comment (jeff-matthews) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:48]:
```suggestion
1. Add the RequireJS config to the `app/design/frontend/<package_name>/<theme_name>/requirejs-config.js` file.
```

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 12]
```diff
@@ -31,6 +31,62 @@ In Magento 2, you have two options for specifying declarative notation:
...
+                    'carousel': 'js/carousel',
+                    '<carousel_name>': 'js/<carousel_name>/<carousel_name>.carousel'
+                }
+            }
+    };
+    ```
+
+You now have two options for specifying declarative notation:
+
+-  use the `data-mage-init` attribute to insert the carousel in a certain element:
```
Comment (jeff-matthews) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:63]:
```suggestion
-  Use the `data-mage-init` attribute to insert the carousel in a certain element:
```

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 13]
```diff
@@ -31,6 +31,62 @@ In Magento 2, you have two options for specifying declarative notation:
...
+
+    ```html
+    <div data-mage-init='{"carousel":{"option": value}}'>
+        <div class="item">Item 1</div>
+        ...
+        <div class="item">Item n</div>
+    </div>
+    ```
+-  use with `<script type="text/x-magento-init"/>`:
```
Comment (jeff-matthews) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:72]:
```suggestion
-  Use with `<script type="text/x-magento-init"/>`:
```

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 14]
```diff
@@ -31,6 +31,62 @@ In Magento 2, you have two options for specifying declarative notation:
...
+
+-  use the `data-mage-init` attribute to insert the carousel in a certain element:
+
+    ```html
+    <div data-mage-init='{"carousel":{"option": value}}'>
+        <div class="item">Item 1</div>
+        ...
+        <div class="item">Item n</div>
+    </div>
+    ```
```
Comment (jeff-matthews) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:71]:
```suggestion
    ```

```

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 15]
```diff
@@ -31,6 +31,62 @@ In Magento 2, you have two options for specifying declarative notation:
...
+
+   ```javascript
+    define(['jquery','<carousel_name>'], function($)
+    {
+        return function(config, element)
+        {
+            $(element).<carousel_name>(config);
+        };
+    });
+   ```
```
Comment (jeff-matthews) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:47]:
```suggestion
   ```

```

**Review by jeff-matthews — CHANGES_REQUESTED** [review]
Thanks @sathiyaaa. Please address my editorial suggestions and I'll merge this.

**Review by jeff-matthews — APPROVED** [review]

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**